builds-cleaner.groovy

```diff
@@ -0,0 +1,30 @@
+/*** BEGIN META {
+  "name" : "Builds Cleaner",
+  "comment" : "Delete all builds keeping ${MAX_BUILDS} from all job of the Jenkins Instance. It checks build is not building and it also keeps the keep forever, lastStableBuild, lastSuccessfulBuild, lastSuccessfulBuild, lastUnstableBuild, lastUnsuccessfulBuild in case are not included in the ${MAX_BUILDS}.",
+  "parameters" : [],
+  "core": "2.107.3",
+  "authors" : [
+    { name : "Marco Davalos" }
+  ]
+} END META**/
+
+import jenkins.model.Jenkins
+import hudson.model.Job
+
+def MAX_BUILDS = 5 // max builds to keep
+
+Jenkins.instance.allItems.findAll { it instanceof hudson.model.Job }.each { job ->
```

Comment:
You already import `hudson.model.Job` so there's no need for you to declare the full package here.

I recommend updating this to:

```groovy
Jenkins.instance.getAllItems(Job.class).each { job ->
...
```

`getAllItems()` method will also find jobs which exist inside of Jenkins folders and sub-folders (i.e. every job on the entire Jenkins system).